doc/design/inference.md

```diff
@@ -0,0 +1,105 @@
+# Design Doc: Inferencer
+
+In fluid, a nueral network is represented as a protobuf message [ProgramDesc](https://github.com/PaddlePaddle/Paddle/blob/develop/doc/design/program.md), the python wrapper of which is `Program`.
+Given a `ProgramDesc`, it can be run on any execution environment.
+In fluid, we call the execution environment `Runtime`, which includes `Place`, `Scope` and `Executor`.
+
+## Representation of the Inference Network
+
+In python, an inference network is defined as:
+
+```python
+image = fluid.layers.data(name='x', shape=[784], dtype='float32')
+predict = fluid.layers.fc(input=image,
+                          size=10,
+                          act='softmax')
+```
+
+After training for serval passes, the parameters can be saved use `fluid.io.save_inference_model`, which will save the binary proto string of the network at the same time.
+```python
+fluid.io.save_inference_model(
+                "./inference_model/", ["x"], [predict],
+                exe)

```

Comment:
- exe->executor?
- Can you give a source code link to `save_inference_model` (since we don't have io in http://www.paddlepaddle.org/docs/develop/documentation/zh/api/v2/fluid.html), in order to explain the meaning of each parameter？Or give a short explanation here? Thus, we can understand why `feed_var_names` and `fetch_var_names` in members come from.

Reply:
Will do in following commits.